smart-contracts/contracts/BridgeBank/BridgeBank.sol

```diff
@@ -574,4 +717,65 @@ contract BridgeBank is BankStorage,
 
         emit LogUnlock(recipient, token, amount);
     }
+
+    /**
+     * @notice Changes the denom of `_token` to `_denom`
+     * @dev Will set the denom both in this contract AND in the token itself
+     * @param _token Address of the BridgeToken
+     * @param _denom The Cosmos denom to be applied
+     * @return true if the operation succeeded
+     */
+    function setBridgeTokenDenom(
+      address _token, string memory _denom
+    ) public onlyOwner returns (bool) {
+      contractDenom[_token] = _denom;
```

Comment:
Let's package up the logic for this function into a function called `_ setBridgeTokenDenom`, then make all the parameters calldata and the function private.

Have `batchSetBridgeTokenDenom` call that function inside it's loop.

Reply:
Then we can make this function external and cut down on gas costs.